xen/drivers/passthrough/iommu.c

```diff
@@ -129,7 +129,7 @@ static void __init parse_iommu_param(char *s)
     } while ( ss );
 }
 
-int iommu_domain_init(struct domain *d)
+int iommu_domain_init(struct domain *d, bool_t use_iommu)
```

Comment:
Yes, I need them for ARM at the moment. But, it should be common. So, my target is *to make ARM happy and not to brake x86*. All I am doing is to call iommu_construct at the domain creation time, but not when the first device is being assigned.